share/functions/__fish_print_hostnames.fish

```diff
@@ -108,7 +108,7 @@ function __fish_print_hostnames -d "Print a list of known hostnames"
             read -z -l contents <$file
 
             # Print hosts from system wide ssh configuration file
-            string replace -rfi '^\s*Host\s+(\S.*?)\s*$' '$1' -- $contents | string split ' ' | string match -v '*\**'
+            string split '\n' -- $contents | string replace -rfi '^\s*Host\s+(\S.*?)\s*$' '$1' | string match -v '*\**'
```

Comment:
This would split on the string `\n`. You don't notice it because the `string replace` will act on each line when reading input from stdin, but if the file contains that string it'll cause weird issues.

Remove the quotes - `string split \n -- $contents`.

Reply:
```
string split '\n' "test'\n'
     '\n'test
     test"
```
return
```
test'
'
'
'test
test
```
---
```
string split \n "test'\n'
     '\n'test
     test"
```
return
```
test'\n'
'\n'test
test
```
Yep, that right.